replica/database.hh

```diff
@@ -1639,8 +1639,8 @@ public:
     // flush all tables in a keyspace on all shards.
     static future<> flush_keyspace_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name);
 
-    static future<> snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush);
-    static future<> snapshot_tables_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, std::vector<sstring> table_names, sstring tag, bool skip_flush);
+    static future<> snapshot_table_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, sstring table_name, sstring tag, bool skip_flush, bool snap_views);
+    static future<> snapshot_tables_on_all_shards(sharded<database>& sharded_db, std::string_view ks_name, std::vector<sstring> table_names, sstring tag, bool skip_flush, bool snap_views);
```

Comment:
nit: Using `bool_class<>` would improve readability of the callers

Reply:
yeah, I followed `skip_flush` convention.
That said, it can also be changed into a bool class.
let's consider that as followup.

Reply:
You already have allow_view_snapshots (which you convert into a bool).

Reply:
see discussion below, its logic is reverse.